numba/typed/typedlist.py

```diff
@@ -377,7 +389,7 @@ def reverse(self):
     def copy(self):
         return _copy(self)
 
-    def index(self, item, start=None, stop=None):
+    def index(self, item: T, start: int = None, stop: int = None) -> int:
```

Comment:
Under strict mode inferred optionals are disabled, so these should be `Optional[int]`

Reply:
thanks, I didn't realize that strict optionals was the default, I thought that inferred optionals were the default.

Reply:
I wonder if we shouldn't consider switching to `--no-strict-optional` to avoid the added verbosity. For example, `index` could fit in one line (easier to read) without losing much in terms of typing (I think `:int = None` is clear enough). what do you think?

```python
    def index(self, item: T, start: pt.Optional[int] = None,
              stop: pt.Optional[int] = None) -> int:
```
vs
```python
def index(self, item: T, start: int = None, stop: int = None) -> int:
```

Reply:
For internal annotations that's something we can decide on.  I think for external facing annotations we should be as precise as possible to avoid incompatibilities for users who don't use `--no-strict-optional`.

Reply:
you're right, I changed it to the explicit `Optional`.